### ISSUE TITLE
SAR: semantic version 2.12.0 not supported anymore? 

### ISSUE THREAD

**Issue by RigoIce** [awslabs/aws-data-wrangler#1098]
**Describe the bug**

According to the documentation for version `2.13.0` one could use 
```typescript
CfnApplication(
    self,
    "wrangler-layer",
    location=CfnApplication.ApplicationLocationProperty(
        application_id="arn:aws:serverlessrepo:us-east-1:336392948345:applications/aws-data-wrangler-layer-py3-8",
        semantic_version="2.12.0",
    ),
)
```
for including the AWS Data Wrangler app from the Serverless Application Repository (SAR). However, if using this example, one gets the following error in the changeset of the CloudFormation stack hosting the `CfnApplication`:
```shell
Status reason
Transform AWS::Serverless-2016-10-31 failed with: Invalid Serverless Application Specification document. Number of errors found: 1. Resource with id [AWSDataWranglerApplication] is invalid. Application arn:aws:serverlessrepo:us-east-1:336392948345:applications/aws-data-wrangler-layer-py3-8 does not have a version 2.12.0
```
It looks like version `2.12.0` is not supported in SAR anymore. The same code works when using `2.13.0` as the semantic version. 

**Environment**

Concerns the version `2.12.0` of the Serverless Application of the Lambda layer. CDK version used was `1.137.0`, as this error appears in CloudFormation the CDK version does not seem to be relevant. 

**To Reproduce**

Use the code above in a CDK stack. The create changeset step will then fail with the error mentioned above.

**Comment by kukushking** [comment]
Hi @RigoIce, the SAR app was re-created last week and 2.12.0 wasn't shared publicly it seems. I have re-published and shared 2.12.0. It might take some time to invalidate cache, but should be available within hours.

**Comment by kukushking** [comment]
Confirmed 2.12.0 & 2.13.0 for Python 3.7 and 3.8 are publicly available; 2.13.0 for Python 3.9 is available as well. Feel free to reopen if the problem persists!
